tests/integration/test_charm.py

```diff
@@ -0,0 +1,144 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test slurmrestd charm against other SLURM charms in the latest/edge channel."""
+
+import asyncio
+import logging
+import pathlib
+import pytest
+import tenacity
+
+from helpers import (
+    get_slurmctld_res,
+    get_slurmd_res,
+)
+from pytest_operator.plugin import OpsTest
+from typing import Any, Coroutine
+
+logger = logging.getLogger(__name__)
+
+SERIES = ["focal"]
+SLURMCTLD = "slurmctld"
+SLURMD = "slurmd"
+SLURMDBD = "slurmdbd"
+SLURMRESTD = "slurmrestd"
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+@pytest.mark.parametrize("series", SERIES)
+async def test_build_and_deploy(
+    ops_test: OpsTest, slurmrestd_charm: Coroutine[Any, Any, pathlib.Path], series: str
+    ) -> None:
+    """Deploy minimal working slurmrestd charm."""
+    res_slurmd = get_slurmd_res()
+    res_slurmctld = get_slurmctld_res()
+
+    charm = await slurmrestd_charm
+
+    await asyncio.gather(
+        # Fetch from charmhub slurmctld
+        ops_test.model.deploy(
+            SLURMCTLD,
+            application_name=SLURMCTLD,
+            channel="edge",
+            num_units=1,
+            resources=res_slurmctld,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            SLURMD,
+            application_name=SLURMD,
+            channel="edge",
+            num_units=1,
+            resources=res_slurmd,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            SLURMDBD,
+            application_name=SLURMDBD,
+            channel="edge",
+            num_units=1,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=SLURMRESTD,
+            num_units=1,
+            series=series,
+        ),
+        ops_test.model.deploy(
+            "percona-cluster",
+            application_name="mysql",
+            channel="edge",
+            num_units=1,
+            series="bionic",
+        ),
+    )
+    
+    # Attach ETCD resource to the slurmctld controller
+    await ops_test.juju("attach-resource", SLURMCTLD, f"etcd={res_slurmctld['etcd']}")
+
+    # Add slurmdbd relation to slurmctld
+    await ops_test.model.relate(SLURMCTLD, SLURMDBD)
+
+    # Add mysql relation to slurmdbd
+    await ops_test.model.relate(SLURMDBD, "mysql")
+
+    # Add slurmctld relation to slurmrestd
+    await ops_test.model.relate(SLURMRESTD, SLURMCTLD)
+
+    # Attach NHC resource to the slurmd controller
+    await ops_test.juju("attach-resource", SLURMD, f"nhc={res_slurmd['nhc']}")
+
+    # Add slurmctld relation to slurmd
+    await ops_test.model.add_relation(SLURMD, SLURMCTLD)
+
+    # issuing dummy update_status just to trigger an event
```

Comment:
Same comment as https://github.com/canonical/slurmd-operator/pull/8#discussion_r1085890460. The word "dummy" fails the inclusive naming check. I recommend applying the same suggestion from the slurmd integration test pull request:

```suggestion
    # Reduce the update status frequency to accelerate the triggering of deferred events.
```